docs/01-app/01-getting-started/07-server-and-client-components.mdx

```diff
@@ -0,0 +1,494 @@
+
+Once a file is marked with `"use client"`, **all its imports and child components are considered part of the client bundle**. This means you don't need to add the directive to every component that is intended for the client.
+
+### Reducing JS bundle size
+
+To reduce the size of your client JavaScript bundles, add `'use client'` to specific interactive components instead of marking large parts of your UI as Client Components.
+
+For example, the `<Layout>` component contains mostly static elements like a logo and navigation links, but includes an interactive search bar. `<Search />` is interactive and needs to be a Client Component, however, the rest of the layout can remain a Server Component.
+
+```tsx filename="app/ui/search.tsx" highlight={1} switcher
+'use client'
+
+export default function Search() {
+  // ...
+}
+```
+
+```tsx filename="app/ui/search.tsx" highlight={1} switcher
+'use client'
+
+export default function Search() {
+  // ...
+}
+```
+
+```tsx filename="app/layout.tsx" switcher
+// Client Component
+import Search from './search'
+// Server Component
+import Logo from './logo'
+
+// Layout is a Server Component by default
+export default function Layout({ children }: { children: React.ReactNode }) {
+  return (
+    <>
+      <nav>
+        <Logo />
+        <Search />
+      </nav>
+      <main>{children}</main>
+    </>
+  )
+}
+```
+
+```jsx filename="app/layout.js" switcher
+// Client Component
+import Search from './search'
+// Server Component
+import Logo from './logo'
+
+// Layout is a Server Component by default
+export default function Layout({ children }) {
+  return (
+    <>
+      <nav>
+        <Logo />
+        <Search />
+      </nav>
+      <main>{children}</main>
+    </>
+  )
+}
+```
+
+### Passing data from Server to Client Components
+
+Data passed as props from Server Components to Client Components need to be [serializable](https://react.dev/reference/react/use-server#serializable-parameters-and-return-values) by React.
+
+If your Client Components depends on data, you can:
+
+- Fetch data on the client using a [community library](/docs/app/getting-started/fetching-data#community-libraries).
+- Fetch data on the server using a [Route Handler](/docs/app/api-reference/file-conventions/route).
+- [Stream data](/docs/app/getting-started/fetching-data#streaming-data-with-the-use-hook) from a Server Component to a Client Component with the [`use` Hook](https://react.dev/reference/react/use).
+
+### Interleaving Server and Client Components
+
+You can pass Server Components as a prop to a Client Component. A common pattern is to use `children` to create a _slot_ in your Client Component.
+
+For example, the `<ClientComponent>` accepts a `children` prop:
+
+```tsx filename="app/client-component.tsx" switcher highlight={1,11}
+'use client'
+
+import { useState } from 'react'
+
+export default function ClientComponent({
```

Comment:
I suggest a more realistic and concrete example to explain this pattern. A counter with a hole is too abstract and will not land for a new reader imo.